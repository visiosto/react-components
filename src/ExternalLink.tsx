// Copyright (c) 2021 Visiosto oy
// Licensed under the MIT License

import React from 'react';

import Icon from './Icon';

type ExternalLinkProps = {
  children: React.ReactNode;
  to: string;
};

function ExternalLink({ children, to }: ExternalLinkProps) {
  return (
    <a href={to} rel="noopener noreferrer" target="_blank">
      {children} <Icon icon="open_in_new" size={18} />
    </a>
  );
}

export default ExternalLink;
