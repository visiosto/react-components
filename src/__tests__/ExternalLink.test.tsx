// Copyright (c) 2021 Visiosto oy
// Licensed under the Apache License, Version 2.0

import React from 'react';

import ExternalLink from '../ExternalLink';

import renderWithProviders from '../util/test/renderWithProviders';

describe('External link component', () => {
  it('renders correctly', () => {
    const { container, getByText } = renderWithProviders(
      <ExternalLink to="https://www.visiosto.fi">Linkki</ExternalLink>,
    );

    expect(container).toMatchSnapshot();

    expect(getByText('Linkki')).toBeInTheDocument();
  });
});
