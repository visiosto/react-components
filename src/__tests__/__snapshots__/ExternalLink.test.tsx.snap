// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`External link component renders correctly 1`] = `
<div>
  <a
    href="https://www.visiosto.fi"
    rel="noopener noreferrer"
    target="_blank"
  >
    Linkki
     
    <span
      class="Icon__Span-sc-1qcplrz-0 dAQUyr material-icons md-18"
      size="18"
    >
      open_in_new
    </span>
  </a>
</div>
`;
